src/Cache/RedisCacheTagsChecksum.php

```diff
@@ -97,16 +95,27 @@ public function isValid($checksum, array $tags) {
   public function calculateChecksum(array $tags) {
     $checksum = 0;
 
-    $fetch = array_values(array_diff($tags, array_keys($this->tagCache)));
-    if ($fetch) {
-      $keys = array_map(array($this, 'getTagKey'), $fetch);
-      foreach ($this->client->mget($keys) as $index => $invalidations) {
-        $this->tagCache[$fetch[$index]] = $invalidations ?: 0;
+    foreach ($tags as $tag) {
+
+      $current = $this->client->get($this->getKey(['tag', $tag]));
+
+      if (!$current) {
+        // Tag has never been created yet, so ensure it has an entry in Redis
+        // database. When dealing in a sharded environment, the tag checksum
+        // itself might have been dropped silently, case in which giving back
+        // a 0 value can cause invalided cache entries to be considered as
+        // valid back.
+        // Note that doing that, in case a tag key was dropped by the holding
+        // Redis server, all items based upon the droppped tag will then become
+        // invalid, but that's the definitive price of trying to being
+        // consistent in all cases.
+        $current = $this->getNextIncrement();
+        $this->client->set($this->getKey(['tag', $tag]), $current);
```

Comment:
Note that I want to restore the tag cache around this, but I'd prefer to add it with a decorator around this class such as *UnreliableRedisCacheTagsChecksum* or *StaticCacheTagsChecksum* register both existing (cached and non cached) into services and let the user choose among those two, I don't care about the one to use per default.